### ISSUE TITLE
refactor(website): documentation website redesign

### ISSUE THREAD

**Issue by d1onys1us** [taikoxyz/taiko-mono#13985]

**Comment by 0xPhoenixeth** [comment]
Add a Quest tab on the site as on Optimism. Increase the level of community engagement with these quests.

**Comment by d1onys1us** [comment]
> Add a Quest tab on the site as on Optimism. Increase the level of community engagement with these quests.

any thoughts on this @MarcusWentz ?

**Comment by 2manslkh** [comment]
If we are going for questing we could do something with [Zealy](https://zealy.io)

**Comment by dantaik** [comment]
Any decision on this one?

**Comment by d1onys1us** [comment]
> Any decision on this one?

yes but this is a bit lower priority (still in q3 though). for now, we will just enhance nextra.

in terms of documentation framework, i'd like something more flexible that supports svelte ideally (keeping an eye on https://github.com/withastro/starlight). i don't want to move to docusaurus or vitepress if they do not support all the features we want, chiefly:
- dynamically rendering child pages in the parent (docusaurus does this, vitepress is hacky with plugins)
- permalinks to docs
- transclusion with docs (mdx supports this, but i'd like a more powerful syntax that supports inheriting just sections of docs)

in terms of our website, we will do a landing page at taiko.xyz, and docs at docs.taiko.xyz. this website will simply use svelte.

so basically, this is on hold. cc @2manslkh @MarcusWentz i have been active in the astro discord to try to bring some of these features to starlight. in the meantime, been opening issues in the nextra repo. so i think we can work with what we have, and move once we have a really solid framework to commit to.
